### PR TITLE
Custom scopes can be browsed in Calypso

### PR DESCRIPTION
**Pull request by carolahp**
Currently, custom scopes, these are scopes created by users through the scopes editor, cannot be correctly browsed on Calypso.
This PR correctly enables the previous feature.

However, creating/editing/removing classes, methods, etc in the scope has not been tested, and remains as work to be done.

This work depends on [NewTools PR#783](https://github.com/pharo-spec/NewTools/pull/783)

### CODE REVIEW

File: src/Calypso-SystemQueries/ClyScopedSystemEnvironment.class.st [hunk 1]
```diff
@@ -90,7 +101,12 @@ ClyScopedSystemEnvironment >> extractPackageNameFrom: aDefinitionString [
 	^ (CDFluidClassDefinitionParser parse: aDefinitionString) packageName
 ]
 
-{ #category : 'accessing' }
+{ #category : 'comparing' }
+ClyScopedSystemEnvironment >> identityHash [
+	^ self scope label identityHash
```
Comment (carolahp) [inline src/Calypso-SystemQueries/ClyScopedSystemEnvironment.class.st]: @tesonep I implemented this to be able to use these system environments as keys in the dictionary `defaultGlobalEnvironments`. It doesn't smell good, but I couldn't think of another way, could you please give me your feedback on it?

Comment (tesonep) [reply src/Calypso-SystemQueries/ClyScopedSystemEnvironment.class.st]: It is strange that the GlobalEnvironment is an identityDictionary. 
I will put a comment and mark it as something we need to fix

**Review by tesonep — CHANGES_REQUESTED** [review]
Can you add a comment in the smelly method

**Comment by guillep** [comment]
Hi @carolahp this is breaking the latest build in a way that it's difficult to work.
Calypso, browse senders and implementors break with a DNU.

I'm going to propose a revert, do not hesitate to redo the PR with a new enhanced  version!

**Comment by carolahp** [comment]
Hello @guillep, I understand, I will take a look now and redo the PR

**Comment by guillep** [comment]
Thanks @carolahp !

**Pull request merged**